### PR TITLE
Plugin works again

### PR DESCRIPTION
**Pull request by KjellConnelly**
I've seen this plugin, but been unable to use it for awhile now. So decided to fix it up. Here's what I have changed:

1. Fixed some bugs that made plugin not work at all
2. Added `Periodic Notes` as a possible alternative to `Daily notes`
3. Added `obsidian-daily-notes-interface` as a dependency. This is a dev library created by Liam Cain (author of the `obsidian-calendar-plugin` and `obsidian-periodic-notes`. This library helps integrate commands, regardless of whether a person is using `Periodic Notes` or `Daily notes`. Removed some of your code as well.
4. Date format is now based on the daily notes format, formatted by `moment.js`. Removed `getISOFormattedDate()`.
5. Added new Settings and updated how they are loaded/saved. Defaults and new keys are easier to add now should anyone choose to add this.
6. New settings include: `boolean` Delete todos from previous day and `boolean` remove empty todos in rollover.
7. Added a new command (accessible from Command Palette) to run rollover (as opposed to only when new files are created). `MAX_TIME_SINCE_CREATION` is ignored in this case.
8. Added new `Notice`s to tell the user what rollover has done (# of todos rolled over, and # of empty todos removed)
9. Updated some config files
10. Improved `README.md` to give users a more complete picture of what the plugin does.

### CODE REVIEW

**Comment by KjellConnelly** [comment]
Ehhh, it doesn't work yet... just realized it didn't compile `obsidian-daily-notes-interface`. Worked on dev mode, not build. Will look at that. Get back to you soon.

**Comment by KjellConnelly** [comment]
Alright it's working now. Give it a shot!

File: package.json [hunk 1]
```diff
@@ -1,19 +1,23 @@
...
-  "version": "0.9.7",
-  "description": "This plugin rolls over incomplete TODOs to the next daily note for Obsidian (https://obsidian.md)",
+  "version": "1.1.0",
+  "description": "This Obsidian.md plugin rolls over incomplete TODOs from the previous daily note to today's daily note. (https://obsidian.md)",
   "main": "main.js",
   "scripts": {
     "dev": "rollup --config rollup.config.js -w",
-    "build": "rollup --config rollup.config.js"
+    "build": "rollup --config rollup.config.js --environment BUILD:production",
+    "deployLocally": "rollup --config rollup.config.js --environment BUILD:production && cp main.js ../../../../My_Obsidian/.obsidian/plugins/obsidian-rollover-daily-todos/main.js"
```
Comment (shichongrui) [inline package.json:9]: This looks specific to your obsidian location relative to the location of this plugin. Could you instead pass the path to obsidian as an argument here?

File: versions.json [hunk 2]
```diff
@@ -2,5 +2,6 @@
 	"1.0.0": "0.9.20",
 	"1.0.1": "0.9.20",
 	"1.0.2": "0.9.20",
-	"1.0.3": "0.12.12"
+	"1.0.3": "0.12.12",
+	"1.0.4": "0.12.12"
```
Comment (shichongrui) [inline versions.json:6]: Version above was changed to 1.1.0 but have 1.0.4 here.

File: index.js [hunk 3]
```diff
@@ -32,46 +43,122 @@ export default class RolloverTodosPlugin extends Plugin {
...
+
+		this.registerEvent(this.app.vault.on('create', async (file) => {
+			this.rollover(file)
 		}))
+
+		this.addCommand({
+      id: "obsidian-rollover-daily-todos",
+      name: "Rollover Todos Now",
+      callback: ()=> this.rollover()
+    })
```
Comment (shichongrui) [inline index.js:161]: Formatting here is off.

**Comment by KjellConnelly** [comment]
Alright I found a few more issues I need to fix before this can go live. 

1. Need to write to end of file if no template header is found
2. Need to create an `undo` command and update Notice to let users know it can be undone. 
3. Seems to have problems on my iPhone. Need to debug this. 
4. Maybe eventually figure out how to handle `read` from disk/cache (I think this is the issue) so unsaved markdown content gets prioritized to modify (if you run command too quickly, auto save doesn't kick in, and todos can get rolled over incorrectly).

File: index.js (hunk 3 above)
Comment (KjellConnelly) [reply index.js]: Just beautified my code in `atom`.

File: package.json (hunk 1 above)
Comment (KjellConnelly) [reply package.json]: Forgot to remove this. But I just removed it, and added a new js file to the `.gitignore` which does the same thing, and of course added that to my local repo.

File: versions.json (hunk 2 above)
Comment (KjellConnelly) [reply versions.json]: Removed this change. Was thinking ahead.

**Comment by KjellConnelly** [comment]
Ok I just fixed 3 of those 4 issues (didn't fix #4, but it seems operable).

1. If no template header is found (Image the case where someone manually creates a new file with the right daily note formatting and puts it in their daily notes folder, but it doesn't have the right heading. Or imagine they change their template after they've set their template heading in this plugin's SettingsTab).
2. A new Undo command was added to the Command Palette. It is only visible for 2 minutes after a rollover has completed. When run, it pulls up a modal, giving a brief description of what will happen, and the number of lines changed in each file (2 files if `delete on complete` is enabled, or just 1 file otherwise). They can then click the cancel button, or confirm the Undo command.
3. It turns out that the function `getLastDailyNote()` was trying to sort by date, but it actually wasn't. I think not many people realized this because when you called `this.app.vault.getAllLoadedFiles()`, it already returned an array of sorted `tfile` objects. But for whatever reason (even though I use the same Vault on my mac, and iPhone), `...getAllLoadedFiles()` would return an array of `tfile`s that were sorted in some other configuration. It seemed random... but maybe it was by file size, or hash value, or something else. I don't know. Anyways, I used `moment.js` to sort. 

Then I realized that the plugin doesn't sort out daily notes from the future. This was never a problem before because even if someone manually created a daily note for the future, todos would just be rolled over to the chronologically latest file. Some may even think it's expected behavior. But with all the new code I added, especially with deleting and what not, and the new undo command, this became a bigger problem. Todos would be rolled, like before, to the last daily note, from the second to last one. But then if `delete on complete` was enabled, todos would be deleted from the daily note from **yesterday**, which would be different from the **second to last** daily note. Anyways, it became a mess. So I just decided to remove future daily notes from the list of daily notes to sort through.

Works on my iPhone now! Finally!

**Pull request merged**